tests/specs/functional/lemmas-spec.k

```diff
@@ -403,6 +403,12 @@ module LEMMAS-SPEC
           ) ... </k>
       requires lengthBytes(B) <Int 2 ^Int 64
 
+    claim [chop-additional-knowledge]: <k> runLemma (
+                      X <=Int chop ( X +Int C ) andBool Y <=Int chop ( Y +Int C )
+              andBool notBool ( ( X +Int C <Int pow256 ) andBool ( Y +Int C <Int pow256 ) )
+          ) => doneLemma ( false ) ... </k>
+    requires #rangeUInt(256, X) andBool #rangeUInt(256, Y) andBool #rangeUInt(256, C)
+
```

Comment:
This is a better version and this is what we essentially run into in the proof. Again, it's tricky to create a representative lemma since the correct way of testing it would be with and without Z3.